### PR TITLE
add redux persist and app level cache

### PR DESCRIPTION
**Pull request by tyler-whitman**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -82,6 +82,8 @@ const App = () => {
     'expo-permissions is now deprecated',
   ])
 
+  LogBox.ignoreAllLogs()
```
Comment (matthewcarlreetz) [inline src/App.tsx:85]: We should remove this. Are there lots of errors now regarding non-serializable objects?

**Review by matthewcarlreetz — COMMENTED** [review]
One thing I noticed is that when you select a filter on the activity tab it will persist on next app launch. I don't think we want that? We might want to blacklist `activitySlice.filter`.

**Comment by matthewcarlreetz** [comment]
I'd like to explore if there are API calls that we can eliminate altogether now that we're caching. I created one example here...
https://github.com/helium/hotspot-app/pull/1082

**Comment by matthewcarlreetz** [comment]
I'm still curious if caching activity in the wallet API would be a good idea

File: src/App.tsx (hunk 1 above)
Comment (tyler-whitman) [reply src/App.tsx:85]: Whoops forgot to push my commit to remove this. Was just using it for testing with offline network

File: src/utils/useCurrency.ts [hunk 2]
```diff
@@ -140,16 +140,25 @@ const useCurrency = () => {
...
+          let [intStr, decStr] = balance
             .toString(undefined, {
               decimalSeparator,
               groupSeparator,
               showTicker: false,
             })
             .split(decimalSeparator)
 
+          // when there is no network the toString method from helium js may not work properly
+          if (intStr === '[object Object]') {
```
Comment (matthewcarlreetz) [inline src/utils/useCurrency.ts:155]: should we fix helium-js?

**Review by matthewcarlreetz — APPROVED** [review]

File: src/utils/useCurrency.ts (hunk 2 above)
Comment (tyler-whitman) [reply src/utils/useCurrency.ts:155]: Its not helium-js its that the object that gets persisted doesnt get the toString method properly so it just ends up returning `[object Object]` when its loaded from cache

**Pull request merged**